### PR TITLE
openshift/origin: add tests for metal ipi platform

### PR DESCRIPTION
**Pull request by andfasano**
This tests could be useful to verify that adding new e2e tests (in particular for the baremetal platform) will pass also on the metal ipi platform.

See also https://github.com/openshift/origin/pull/25853

### CODE REVIEW

**Comment by ardaguclu** [comment]
Does that trigger when new PR is opened to master from another branch in origin?, or after being merged to master.

**Comment by andfasano** [comment]
> Does that trigger when new PR is opened to master from another branch in origin?, or after being merged to master.

Presubmit jobs are triggered before

**Comment by stbenjam** [comment]
Will you update the test list to include `Feature:baremetal` or wait until the PR is merged? I thought it'd be fine to include a match for a test that might not exist yet

**Comment by andfasano** [comment]
> Will you update the test list to include `Feature:baremetal` or wait until the PR is merged? I thought it'd be fine to include a match for a test that might not exist yet

I was waiting for this PR to get merged first, but effectively I can update the test lists meanwhile, it should be fine

**Comment by sttts** [comment]
What does "will pass also on the metal ipi platform." mean?

File: ci-operator/config/openshift/origin/openshift-origin-master.yaml [hunk 1]
```diff
@@ -182,6 +186,38 @@ tests:
...
+    env:
+      DEVSCRIPTS_CONFIG: |
+        IP_STACK=v4v6
+        NETWORK_TYPE=OVNKubernetes
+    workflow: baremetalds-e2e
+- artifact_dir: /tmp/artifacts
+  as: e2e-metal-ipi-virtualmedia
+  steps:
+    cluster_profile: packet
+    workflow: baremetalds-e2e-virtualmedia
```
Comment (sttts) [inline ci-operator/config/openshift/origin/openshift-origin-master.yaml:220]: these should be optional to start with, until proven to be stable.

Comment (andfasano) [reply ci-operator/config/openshift/origin/openshift-origin-master.yaml:220]: It's a good point. Usually in the other repos where we added those tests they are all option but `e2e-metal-ipi-ovn-ipv6` (note: some of these are also gating the release), I could follow the same approach?

**Comment by andfasano** [comment]
> What does "will pass also on the metal ipi platform." mean?

The basic idea is that whenever there will be the need of adding a new e2e test for the baremetal platform (like the two in the https://github.com/openshift/origin/pull/25853) it could be possible to verify in the PR that the newly added tests:

* Pass successfully on any e2e-metal-ipi-* job
* Are skipped on all the other jobs

File: ci-operator/config/openshift/origin/openshift-origin-master.yaml (hunk 1 above)
Comment (andfasano) [reply ci-operator/config/openshift/origin/openshift-origin-master.yaml:220]: e2e-metal-ipi (IPv4), e2e-metal-ipi-dualstack and e2e-metal-ipi-virtualmedia marked as optional.

**Comment by andfasano** [comment]
@sttts Above failures look to be not related to the e2e-metal-ipi jobs, which succeeded

File: ci-operator/config/openshift/origin/openshift-origin-master.yaml [hunk 2]
```diff
@@ -3,6 +3,10 @@ base_images:
     name: "4.8"
     namespace: ocp
     tag: base
+  dev-scripts:
+    name: test
+    namespace: ocp-kni
+    tag: dev-scripts
```
Comment (sttts) [inline ci-operator/config/openshift/origin/openshift-origin-master.yaml:9]: for what is this used?

Comment (stbenjam) [reply ci-operator/config/openshift/origin/openshift-origin-master.yaml:9]: Development environment for baremetal

Comment (andfasano) [reply ci-operator/config/openshift/origin/openshift-origin-master.yaml:9]: And without that image all the e2e-metal-ipi workflows couldn't work (see [1])

[1] https://github.com/openshift/release/blob/662bbc163381c6af0f48d8488444c9f9707069d1/ci-operator/step-registry/baremetalds/devscripts/setup/baremetalds-devscripts-setup-ref.yaml#L3

**Comment by stbenjam** [comment]
@stts Can you take another look please?

**Comment by stbenjam** [comment]
/assign @sttts

Could you have another look please?

**Comment by adambkaplan** [comment]
@andfasano was the `metal-ipi-ovn-ipv6` test intended to be run on _every_ PR in origin? Or are these an optional test?

The current job history suggests that this test is very flaky and is not suitable to block merges in origin.

https://prow.ci.openshift.org/job-history/gs/origin-ci-test/pr-logs/directory/pull-ci-openshift-origin-master-e2e-metal-ipi-ovn-ipv6?buildId=

**Pull request merged**